### PR TITLE
Add Resource rule for file and browseobject contexts

### LINKED ISSUE #3620: Graphics files in .net standard 2.0 library do have have build action of "Resource"
We have a few graphics files in .net standard 2.0 libraries, and .net 4.7.1 libraries. We are using     
MSBuildSdkExtras in these libraries. All the libraries have been converted to the new project system. However, when we try to set the build action to "Resource", on an image file like png or gif, it 
is not an available option in .net standard 2.0 libraries.

### PR DESCRIPTION
**Pull request by davidwengier**
This change allows selecting Resource as a build action in the properties dialog and ensures that is saved to the project file correctly. 

Fixes #3620

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/Microsoft.VisualStudio.ProjectSystem.Managed.csproj [hunk 1]
```diff
@@ -263,6 +266,12 @@
       <Generator>MSBuild:GenerateRuleSourceFromXaml</Generator>
       <SubType>Designer</SubType>
     </XamlPropertyRule>
+    <XamlPropertyRuleNoCodeBehind Include="ProjectSystem\Rules\Resource.BrowseObject.xaml" />
+    <XamlPropertyRule Include="ProjectSystem\Rules\Resource.xaml">
+      <XlfInput>false</XlfInput>
```
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/Microsoft.VisualStudio.ProjectSystem.Managed.csproj]: Why? The BrowseObject has strings.

Comment (davkean) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/Microsoft.VisualStudio.ProjectSystem.Managed.csproj]: Ugh, mb - this isn't the browseobject.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/Resource.BrowseObject.xaml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    <StringProperty.DataSource>
+      <DataSource PersistenceStyle="Attribute" SourceOfDefaultValue="AfterContext" />
+    </StringProperty.DataSource>
+  </StringProperty>
+  <StringProperty Name="Extension" Visible="False" ReadOnly="true">
+    <StringProperty.DataSource>
+      <DataSource Persistence="Intrinsic" ItemType="Resource" PersistedName="Extension" SourceOfDefaultValue="AfterContext" />
+    </StringProperty.DataSource>
+  </StringProperty>
+  <StringProperty Name="LastGenOutput" Visible="false" Description="The filename of the last file generated as a result of the SFG." />
```
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/Resource.BrowseObject.xaml]: For future reference during cleanup, we don't need descriptions for non-visible properties.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/ProjectItemsSchema.xaml [hunk 3]
```diff
@@ -66,12 +66,15 @@
 
     <ContentType Name="XamlAppDef" DisplayName="Workflow definition" ItemType="XamlAppDef" />
 
+    <ContentType Name="Resource" DisplayName="Resource" ItemType="Resource" />
```
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/ProjectItemsSchema.xaml]: I wonder aloud if we want to go back and give these things better names to describe what they are used for; Resource (WPF), Application definition (WPF)?

Comment (davidwengier) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/ProjectItemsSchema.xaml]: I was wondering that too but at least with the Resource case I seem to remember using it in Winforms at least once. From memory it made it an embedded resource with no namespace prefix (as distinct from Embedded Resource), but maybe it doesn't matter if it says WPF in the name anyway

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/Resource.BrowseObject.xaml (hunk 2 above)
Comment (davidwengier) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/Resource.BrowseObject.xaml]: 👍 Makes sense. Probably more I can clean up in the other PR then too

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/ProjectItemsSchema.xaml (hunk 3 above)
Comment (davkean) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/ProjectItemsSchema.xaml]: I don't believe this used by WinForms, or if you used it in WinForms, it was via the WinFx targets. See FileClassification target/FileClassifier task which handles these in Microsoft.WinFx.targets.

Comment (davkean) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/ProjectItemsSchema.xaml]: You might be thinking of csc's `-resource` switch, which is how `<EmbeddedResource/>` indirectly ends up in the assembly.

Comment (davidwengier) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/ProjectItemsSchema.xaml]: I must be remembering wrong, I've just checked and the Resource item type does nothing in a standard WinForms app.

**Pull request merged**